### ISSUE TITLE
Output to directory that doesn't exist doesn't work

### ISSUE THREAD

**Issue by bwspenc** [idaholab/moose#3249]
If I put something like this in my Outputs block:
```
[./CSV]
  type=CSV
  file_base = csv/mybase
[../]
```
the code dies with an error because the 'csv' directory doesn't exist.  Any chance we could have MOOSE make that directory for the user if it doesn't exist?

**Comment by aeslaughter** [comment]
Nobody seems to really want this behavior or to persuade @friedmud since this has been sitting around 6 months, so I am closing it.

**Comment by bwspenc** [comment]
This got closed a while ago, but this continues to bite us. I just ran into this today while teaching a new user, so I'm bringing this up again.

We have models that dump out a large number of csv files, so it's pretty painful to dump those all in the directory where the job is run. Hence, we do something like 'file_base = csv/out'. It's really easy to forget to add that csv directory before you run the job, and then the job fails after being in the queue for a while because of this. The fewer failure points and glitches like this to explain to users, the better.

So, I'm going to submit a PR shortly that fixes this issue in the hopes that I can convince someone that this is a good idea...
